import "goby/moos/protobuf/frontseat_config.proto";
import "goby/common/protobuf/option_extensions.proto";

message ABCFrontSeatConfig
{
    required string tcp_address = 1;
    optional uint32 tcp_port = 2 [default = 54321];
    message StartParams
    {
        required double lat = 1;
        required double lon = 2;
        required int32 duration = 3;
    }
    required StartParams start = 3;
}

extend iFrontSeatConfig
{
    optional ABCFrontSeatConfig abc_config = 1001;
}
